File: Selection statement/Question3/main.cpp
#include <iostream>

int StudentDiscount(int n)
{
    float discount;

    if (n <= 500)
    {
        discount = (5 * n) / 100;
        std::cout << "Your discount amount is: " << discount << std::endl;
    }
    else
    {
        discount = (10 * n) / 100;
        std::cout << "Your discount amount is: " << discount << std::endl;
    }
}

int NonStudentDiscount(int n)
{
    float discount;

    if (n <= 500)
    {
        discount = (2 * n) / 100;
        std::cout << "Your discount amount is: " << discount << std::endl;
    }
    else
    {
        discount = (8 * n) / 100;
        std::cout << "Your discount amount is: " << discount << std::endl;
    }
}

int main()
{
    char response;

    std::cout << "Are you a student? (y/n): ";
    std::cin >> response;

    if (response == 'y' || response == 'Y')
    {
        int n;
        std::cout << "Enter your bill amount: ";
        std::cin >> n;
        StudentDiscount(n);
    }
    else if (response == 'n' || response == 'N')
    {
        int n;
        std::cout << "Enter your bill amount: ";
        std::cin >> n;
        NonStudentDiscount(n);
    }
    else
    {
        std::cout << "Invalid input" << std::endl;
    }

    return 0;
}

File: Selection statement/Question3/main.py
def studentDiscount(n):
    if n <= 500:
        discount = ((5/100) * n)
        print("Your discount amount is: ", discount)
    else:
        discount = ((10/100) * n)
        print("Your discount amount is: ", discount)


def nonStudentDiscount(n):
    if n <= 500:
        discount = ((2/100) * n)
        print("Your discount amount is: ", discount)
    else:
        discount = ((8/100) * n)
        print("Your discount amount is: ", discount)


def discount():
    a = input("Are you a student? (y/n): ")

    if a[0] == "Y" or a[0] == "y":
        n = int(input("Enter your bill amount: "))
        studentDiscount(n)
    elif a[0] == "N" or a[0] == "n":
        n = int(input("Enter your bill amount: "))
        nonStudentDiscount(n)
    else:
        print("Invalid input")


discount()
